src/instructlab/model/accelerated_train.py

```diff
@@ -196,6 +196,53 @@ def accelerated_train(
         run_training(train_args=train_args, torch_args=torch_args)
 
 
+def _run_phase(
+    train_args: TrainingArgs,
+    torch_args: TorchrunArgs,
+    data_path: pathlib.Path,
+    checkpoint_dir: pathlib.Path,
+    num_epochs: int | None,
+    samples_per_save: int | None,
+    effective_batch_size: int | None,
+    journal: TrainingJournal,
+    phase_model: TrainPhaseModel | EvalPhaseModel,
+    next_phase: TrainingPhases,
+    model_override: pathlib.Path | None = None,
+):
+    """Runs a single phase of the multi-phase training pipeline and capture any errors.
+
+    Raises:
+        click.exceptions.Exit: Raises a click exception in lieu of a system exit.
+    """
+    exception = None
+    try:
+        _training_phase(
+            train_args=train_args.model_copy(deep=True),
+            torch_args=torch_args,
+            data_path=data_path,
+            checkpoint_dir=checkpoint_dir,
+            num_epochs=num_epochs,
+            samples_per_save=samples_per_save,
+            effective_batch_size=effective_batch_size,
+            model_override=model_override,
+            # model override not necessary because we expect model to come from ctx.params.model_path.
+        )
+    # pylint: disable=broad-except # training library emits a general Exception in lieu of a more specific one.
+    except (RuntimeError, KeyboardInterrupt, Exception) as e:
+        if not isinstance(e, KeyboardInterrupt):
+            logger.error("Failed during training loop: ", e)
+        exception = e
+    else:
```

Comment:
I think this indentation might be wrong? or this might be a try/catch syntax thing I only barely remember

Reply:
This is a try/catch specific syntax. The `else` here is basically saying "if I've caught an exception, handle it in these except blocks, else run this code"

Reply:
++